webapp/loggedin/modules/data/dataVis/dataQuery/components/tableHeader.js

```diff
@@ -57,14 +58,14 @@ class TableHeader extends React.Component {
       updateTableFilter, updateTableOffset, updateTableEditMode,
     } = this.props
 
-    const csvDownloadLink = `/api/surveyRdb/${surveyId}/${tableName}/export?filter=${filter}&sort=${DataSort.serialize(sort)}&cols=${JSON.stringify(colNames)}`
+    const csvDownloadLink = `/api/surveyRdb/${surveyId}/${tableName}/export?filter=${JSON.stringify(filter)}&sort=${DataSort.serialize(sort)}&cols=${JSON.stringify(colNames)}`
     const sortMsg = DataSort.getViewExpr(sort)
 
     return (
       <div className="table__header">
 
         <div className="data-operations">
-          <Tooltip messages={filter && [filter]}>
+          <Tooltip messages={R.isNil(filter) ? [] : [Expression.toString(filter, Expression.modes.sql)]}>
```

Comment:
can you still use 
```messages = {filter && [Expression.toString(filter, Expression.modes.sql)]}``` ?

Reply:
Not anymore, we have to check that filter is not null in tooltip.js to do so

Reply:
@sgiaccio is right, "filter" now can be null

Reply:
but if filter is null, if you do ```messages = {filter && [Expression.toString(filter, Expression.modes.sql)]}```, wouldn't it return null?

Reply:
Yes we can do that, then we need to add an additional control on Tooltip

Reply:
that would be better I think